### PR TITLE
Space travel hub

### PR DESCRIPTION
**Pull request by Bigizi**
**Space-Travel-Hub project**
Space travel hub is a single-page application that displays the rockets, dragons, missions, and profiles
in this project, you are able to make reservations for rocket or dragon
This project has different components and styling
We have the following components: dragon.js, rocket.js, navigation.js, mission.js, and profile.js and these files have the corresponding styles.
We also have tests for each component.

### CODE REVIEW

File: src/components/dragons/Dragon.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+} from '../../redux/Dragons/dragonSlice';
+import './dragon.css';
+
+function Dragon() {
+  const dispatch = useDispatch();
+  const dragons = useSelector(selectAllDragons);
+  useEffect(() => {
+    dispatch(fetchDragons());
+  }, [dispatch]);
+
```
Comment (PinkMoon25) [inline src/components/dragons/Dragon.js:14]: - [ ] Great job implementing the project requirements, however please notice that when I book a rocket or a dragon, and go to my profile it is there, but when I come back to the rocket or dragons page the reserved badge is gone it is also not present under my profile section, I have recorded a small video to make it clear :point_down: 

https://user-images.githubusercontent.com/97521557/225404090-e5c2290a-93a2-4ba7-931d-042903b9e1d0.mp4

Please try to fix this by adding a conditional check like you have done for the missions, in the video you will see the mission when joined is preserved when I visit the mission page or my profile section again.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { fetchRockets } from './redux/Rockets/rocketSlice';
+import Dragon from './components/dragons/Dragon';
+
+function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchMissions());
+    dispatch(fetchDragons());
+    dispatch(fetchRockets());
+  }, [dispatch]);
```
Comment (PinkMoon25) [inline src/App.js:20]: - [ ] Hi Team, As mentioned in the requirements :point_down: 
image

Please make sure that you dispatch the actions only when needed, so fetch rockets when you initially render the page then fetch dragons on dragons page and missions on missions page, remove this useEffect and dispatch from here and only use it inside your components so whenever they render the useEffect will fetch the data, you should add a conditional check also so it does not fetches data on every re-render, you can[ use redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) to keep track of the actions you are dispatching

File: src/redux/Dragons/dragonSlice.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+//   reducers: {},
+//   extraReducers: (builder) => {
+//     builder
+//       .addCase(fetchDragons.fulfilled, (state, action) => action.payload);
+//   },
+// });
+
+// export const selectDragons = (state) => state.dragons;
+
+// export default dragonSlice.reducer;
```
Comment (PinkMoon25) [inline src/redux/Dragons/dragonSlice.js:21]: - [ ] Please remove this commented code as it is mentioned in the best practices, we do not commit old pieces of code to the codebase

File: src/__tests__/dragon.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    render(
+      <Provider store={store}>
+        <Dragon />
+      </Provider>,
+    );
+
+    expect(store.dispatch).toEqual(expect.any(Function));
+  });
+  it('should dispatch bookDragons action on reserve button click', () => {
+    // Test case for bookDragons action creator
```
Comment (PinkMoon25) [inline src/__tests__/dragon.test.js:50]: - [ ] [OPTIONAL] I think you are missing some test case here, and also try to test your reducers as they are pure functions, maybe you can test them for each of the components, as currently you are only testing render part.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+I would like to thank our parents and supporters for their support
+I would like to thank All my coding partners
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (PinkMoon25) [inline README.md:180]: - [ ] Please fix your MIT link as currently it is broken and giving a 404 error

**Review by PinkMoon25 — CHANGES_REQUESTED** [review]
Hi @Bigizi @rivasbolinga @djo1975 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linter checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Code reviews :heavy_check_mark: 
Pair-programming :heavy_check_mark: 
React redux :heavy_check_mark: 
Missions section :heavy_check_mark: 
live data from SpaceX Api :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Bigizi @djo1975  @rivasbolinga , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**